### PR TITLE
Cart management

### PR DESCRIPTION
**Pull request by purvasharmadev**
## Files to Check:

> cart.js
> cart-context.js
> CartCard.jsx
> CartPriceDetail.jsx

### CODE REVIEW

**Review by srishtigautam21 — APPROVED** [review]
very well structured code
Login suggestion:
Give already written dummy login  details so that as a reviewer i can easily login and see all of your fuctionality on your ecom

**Comment by purvasharmadev** [comment]
> very well structured code Login suggestion: Give already written dummy login details so that as a reviewer i can easily login and see all of your fuctionality on your ecom

Thank you for approving the code. 
I have given a "browse as guest" login option, will add a description to it. Thank you for your suggestion.

**Review by adityapandadev — APPROVED** [review]

File: src/Components/Cart/cart.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { Footer } from "../../Pages/Footer";
+import { Link } from "react-router-dom";
+import { Nav } from "../../Pages/Nav";
+import { CartCard } from "../../Pages/CartCard";
+import { CartPriceDetail } from "../../Pages/CartPriceCard";
+import "../../Styles/cart.css";
+import { useCart } from "../../Context/cart-context";
```
Comment (iRohitGaur) [inline src/Components/Cart/cart.jsx:7]: suggestion: anything other than a Component should start with a lower case letter.

File: src/Components/Cart/cart.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          {/* <!-- Product Price Details --> */}
+
+          <div className="card cart-card p-1">
+            <h2 className="order-head">Price Details</h2>
+            <CartPriceDetail />
+          </div>
+        </div>
+      ) : (
+        <div
+          style={{ height: "100vh" }}
```
Comment (iRohitGaur) [inline src/Components/Cart/cart.jsx:49]: suggestion: avoid inline styling. Provide a classname here and write the css in the css file.

File: src/Context/cart-context.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  // Function to remove item from cart
+
+  async function removeFromCart(item) {
+    try {
+      const res = await axios.delete(`api/user/cart/${item._id}`, axiosConfig);
+      setProductCart(res.data.cart);
+    } catch (error) {
+      console.log("error is :", error);
```
Comment (iRohitGaur) [inline src/Context/cart-context.js:52]: suggestion: remove `console.log`

File: src/Context/cart-context.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        0
+      )
+    );
+  }, [productCart]);
+
+  useEffect(() => {
+    if (response != undefined) {
+      getCartProducts();
+    }
+  }, [response]);
```
Comment (iRohitGaur) [inline src/Context/cart-context.js:91]: You are fetching the cart array again on every change in response. This is unnecessary as the mock backend api already gives you the updated cart array in response.

suggestion: this part is not needed. Better to set the `productCart` on those response.

**Comment by iRohitGaur** [comment]
suggestion: you can remove the 0 badge from `wishlist` and `cart` icon and only show a badge when there is an item in them.

**Review by iRohitGaur — APPROVED** [review]
UI looks good.

The code is clean and understandable.

appreciation: good use of comments to segregate appropriate sections

I have given some suggestions. Please look into it.

Good to merge with updates 👍🏻

**Comment by purvasharmadev** [comment]
> UI looks good.
> 
> The code is clean and understandable.
> 
> appreciation: good use of comments to segregate appropriate sections
> 
> I have given some suggestions. Please look into it.
> 
> Good to merge with updates 👍🏻

Thank you for your suggestions, will surely look into it.

**Pull request merged**